src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -59,14 +69,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     String fileName = filePart.getSubmittedFileName();
     InputStream fileInputStream = filePart.getInputStream();
 
-    // Upload the file and get its URL
+    // Upload the image and get its URL
     String imageURL;
     boolean hasImage;
+    ArrayList<String> imageTags = new ArrayList<String>();
     
+    // Upload the image and save metadata if provided
     try{
+        byte[] imageBytes = fileInputStream.readAllBytes();
         imageURL = uploadToCloudStorage(fileName, fileInputStream);
         hasImage = true;
+
+        // Get the labels of the image that the user uploaded.
+        List<EntityAnnotation> imageLabels = getImageLabels(imageBytes);
+        
+        for (EntityAnnotation label : imageLabels) {
+            imageTags.add(label.getDescription());
+        }
     }
+    // Case for no image uploaded
```

Comment:
Put this comment inside the catch branch.

Reply:
Another option is to just have an if to check if fileName.isEmpty(), and then only do this try-catch if it's not empty. Basically, only try to upload if we do have an image. This way we know that the exception means there's actually an error, and we'd probably want to propagate that exception up, instead of swallowing it, does that make sense? so something like:

```
if(!fileName.isEmpty()) {
  try {
     // upload image
  } catch (Exception e) {
   // a real error when uploading.
  throw e;
  } 
}
```